extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -121,22 +118,26 @@ int fill(final long position, final ByteBuffer dest) throws IOException {
             final int impliedReadAhead = (int) (lastFragmentIx - firstFragmentIx);
             final int desiredReadAhead = instructions.readAheadCount();
             final long totalRemainingFragments = numFragments - firstFragmentIx - 1;
-            final int maxReadAhead = instructions.maxCacheSize() - 1;
-            readAhead = Math.min(
-                    Math.max(impliedReadAhead, desiredReadAhead),
-                    (int) Math.min(maxReadAhead, totalRemainingFragments));
+            readAhead = Math.min(Math.max(impliedReadAhead, desiredReadAhead), totalRemainingFragments);
         }
-        final S3Request firstRequest = getOrCreateRequest(firstFragmentIx);
+        // Hold a reference to the first request to ensure it is not evicted from the cache
+        acquiredRequest = getOrCreateRequest(firstFragmentIx);
         for (int i = 0; i < readAhead; ++i) {
+            // Do not hold references to the read-ahead requests
```

Comment:
I worry about high memory pressure situations, and readahead exacerbating it. We should sync up w/ Ryan, and at a minimum file a ticket for future follow up IMO.

Reply:
Will make an issue and link here.

Reply:
https://github.com/deephaven/deephaven-core/issues/5649